homeassistant/components/gogogate2/common.py

```diff
@@ -0,0 +1,130 @@
+"""Common code for GogoGate2 component."""
+from datetime import datetime, timedelta
+import logging
+from typing import Callable, NamedTuple, Optional
+
+from gogogate2_api import GogoGate2Api, InfoResponse
+from gogogate2_api.common import Door, get_configured_doors
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import DATA_MANAGERS, DATA_UPDATED_SIGNAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class StateData(NamedTuple):
+    """State data for a cover entity."""
+
+    config_unique_id: str
+    unique_id: Optional[str]
+    door: Optional[Door]
+
+
+class DataManager:
```

Comment:
This seems like it is reimplementing `DataUpdateCoordinator`. 

https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities